### PR TITLE
Refactors some defines for easier Species Development

### PR DESCRIPTION
**Pull request by Zandario**
## About The Pull Request

Me following out on my promise to help with the upcoming Species, since this codebase isn't well-suited for such things, this here is a step to help with that.

## Why It's Good For The Game

Doesn't really do anything to the game, this is just back-end shit.

## Changelog
:cl:
code: Added some framework for future species expansions, including clothing refitting.
refactor: Made majority of the relevant Species IDs and Categories pre-defined, also for easier expansion and use.
/:cl:

### CODE REVIEW

**Comment by Zandario** [comment]
This is only the first part of what I need to get done, I'll get the dmi part sorted later

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -814,8 +814,8 @@
 
 	if(isjellyperson(H))
 		to_chat(H, "<span class='warning'>Your jelly shifts and morphs, turning you into another subspecies!</span>")
-		var/species_type = pick(subtypesof(/datum/species/jelly))
-		H.set_species(species_type)
+		var/species_category = pick(subtypesof(/datum/species/jelly))
+		H.set_species(species_category)
```
Comment (timothyteakettle) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: i dont think species category is the right change here, assuming you did a replace all on it and it caught this, as it's an entirely different thing to species type

File: code/game/objects/items.dm [hunk 2]
```diff
@@ -156,6 +156,34 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
+
+	//** These specify item/icon overrides for _species_
+
+	/* Species-specific sprites, concept stolen from Paradise//vg/.
+	ex:
+	sprite_sheets = list(
+		SPECIES_FELINID = 'icons/cat/are/bad'
+		)
+	If index term exists and icon_override is not set, this sprite sheet will be used.
+	*/
```
Comment (LetterN) [inline code/game/objects/items.dm]:
```suggestion
	/* 
	 * Species-specific sprites, concept stolen from Paradise//vg/.
	 * ex:
	 * sprite_sheets = list(
	 * 	SPECIES_FELINID = 'icons/cat/are/bad'
	 * 	)
	 * If index term exists and icon_override is not set, this sprite sheet will be used.
	 */
```
dmdoc

File: code/modules/clothing/clothing.dm [hunk 3]
```diff
@@ -49,8 +49,13 @@
...
 	//Add a "exclude" string to do the opposite, making it only only species listed that can't wear it.
 	//You append this to clothing objects
-
-
+	/*
+		Sprites used when the clothing item is refit. This is done by setting icon_override.
+		For best results, if this is set then sprite_sheets should be null and vice versa, but that is by no means necessary.
+		Ideally, sprite_sheets_refit should be used for "hard" clothing items that can't change shape very well to fit the wearer (e.g. helmets, hardsuits),
+		while sprite_sheets should be used for "flexible" clothing items that do not need to be refitted (e.g. aliens wearing jumpsuits).
+	*/
```
Comment (LetterN) [inline code/modules/clothing/clothing.dm]:
```suggestion
	/*
	 * 	Sprites used when the clothing item is refit. This is done by setting icon_override.
	 * 	For best results, if this is set then sprite_sheets should be null and vice versa, but that is by no means necessary.
	 * 	Ideally, sprite_sheets_refit should be used for "hard" clothing items that can't change shape very well to fit the wearer (e.g. helmets, hardsuits),
	 * 	while sprite_sheets should be used for "flexible" clothing items that do not need to be refitted (e.g. aliens wearing jumpsuits).
	 */
```

File: code/modules/clothing/clothing.dm [hunk 4]
```diff
@@ -49,8 +49,13 @@
...
+	var/list/sprite_sheets_refit = null
```
Comment (LetterN) [inline code/modules/clothing/clothing.dm]: you do not need to explicitly set this to null

File: code/modules/clothing/clothing.dm [hunk 5]
```diff
@@ -477,10 +482,53 @@ BLIND     // can't see anything
...
-
 /// If we're a clothing with at least 1 shredded/disabled zone, give the wearer a periodic heads up letting them know their clothes are damaged
 /obj/item/clothing/proc/bristle(mob/living/L)
 	if(!istype(L))
 		return
 	if(prob(0.2))
 		to_chat(L, "<span class='warning'>The damaged threads on your [src.name] chafe!</span>")
+
+
+/obj/item/clothing/proc/refit_for_species(var/target_species)
```
Comment (LetterN) [inline code/modules/clothing/clothing.dm]:
```suggestion
/obj/item/clothing/proc/refit_for_species(target_species)
```
do not add `var/` on args

File: code/modules/clothing/clothing.dm [hunk 6]
```diff
@@ -477,10 +482,53 @@ BLIND     // can't see anything
...
+	// Set icon
+	if (sprite_sheets_refit && (target_species in sprite_sheets_refit))
+		sprite_sheets[target_species] = sprite_sheets_refit[target_species]
+
+	if (sprite_sheets_obj && (target_species in sprite_sheets_obj))
+		icon = sprite_sheets_obj[target_species]
+	else
+		icon = initial(icon)
+
+/obj/item/clothing/head/helmet/refit_for_species(var/target_species)
```
Comment (LetterN) [inline code/modules/clothing/clothing.dm]:
```suggestion
/obj/item/clothing/head/helmet/refit_for_species(target_species)
```

**Review by LetterN — CHANGES_REQUESTED** [review]

File: code/game/objects/items.dm [hunk 7]
```diff
@@ -156,6 +156,35 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
 	var/canMouseDown = FALSE
 
 
+	///Species Sprite Overrides
```
Comment (silicons) [inline code/game/objects/items.dm]: ok so all of the item/clothing stuff can't be merged unless we know exactly what we're doing with it
we have tens of thousands of items per round and these are all wasted lists unless they're specifically needed, even with a typelist system like the rest of the code uses

**Pull request merged**